bokehjs/src/coffee/models/annotations/label_set.coffee

```diff
@@ -9,42 +9,45 @@ class LabelSetView extends TextAnnotation.View
   initialize: (options) ->
     super(options)
 
-    @xmapper = @plot_view.frame.get('x_mappers')[@mget("x_range_name")]
-    @ymapper = @plot_view.frame.get('y_mappers')[@mget("y_range_name")]
+    @xmapper = @plot_view.frame.get('x_mappers')[@model.x_range_name]
+    @ymapper = @plot_view.frame.get('y_mappers')[@model.y_range_name]
 
-    @set_data(@mget('source'))
-    @set_visuals(@mget('source'))
+    @set_data()
 
-    if @mget('render_mode') == 'css'
+    if @model.render_mode == 'css'
       for i in [0...@_text.length]
         @title_div = $("<div>").addClass('bk-annotation-child').hide()
         @title_div.appendTo(@$el)
 
   bind_bokeh_events: () ->
-    if @mget('render_mode') == 'css'
+    if @model.render_mode == 'css'
       # dispatch CSS update immediately
       @listenTo(@model, 'change', () ->
-        @set_data(@mget('source'))
+        @set_data()
         @render())
-      @listenTo(@mget('source'), 'change', () ->
-        @set_data(@mget('source'))
+      @listenTo(@model.source, 'change', () ->
+        @set_data()
         @render())
```

Comment:
I think `@plot_view.request_render()` should work, that would be better than a synchronous call to `render` if so.